### PR TITLE
Add Runtime command tests

### LINKED ISSUE #458: Add test coverage for RuntimeCommands
As a developer, I'd like to add test coverage for `RuntimeCommands`, so I can incrementally validate runtime state of the deployed applications through local builds and CI plans.

### PR DESCRIPTION
**Pull request by ilayaperumalg**
This resolves #458

### CODE REVIEW

File: spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/RuntimeCommands.java [hunk 1]
```diff
@@ -57,9 +57,13 @@
 public class RuntimeCommands implements CommandMarker {
 
 	private static final String LIST_MODULES = "runtime modules";
+	
+	private DataFlowShell dataFlowShell;
```
Comment (markfisher) [inline spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/RuntimeCommands.java]: other than the setter, this appears to be unused

File: spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/RuntimeCommands.java [hunk 2]
```diff
@@ -57,9 +57,13 @@
...
 
 	@Autowired
-	private DataFlowShell dataFlowShell;
+	public void setDataFlowShell(DataFlowShell dataFlowShell) {
```
Comment (markfisher) [inline spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/RuntimeCommands.java]: a constructor might be better... and could we even avoid autowiring?

Comment (ilayaperumalg) [inline spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/RuntimeCommands.java]: I believe this is needed only for testing purpose where we use mocks and the `DataFlowShell` isn't available in the context already. I think we would need to `autowire` the `DataFlowShell` anyway as all the commands are scanned as components, is it not the case?

Comment (markfisher) [inline spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/RuntimeCommands.java]: ultimately I think we should avoid scanning, but that's part of a bigger effort

I'm just never a fan of seeing a change that's made purely for the tests

**Comment by markfisher** [comment]
rebased and merged as https://github.com/spring-cloud/spring-cloud-dataflow/commit/b25146d288cbdc51b6ec2f72ebdd6a3158ce9a55

along with: https://github.com/spring-cloud/spring-cloud-dataflow/commit/982ca69930fe6e52eeb9dd54b3375c83ea7a69ef

**Pull request closed without merging**